gozo-stores/src/functions/functions.js

```diff
@@ -0,0 +1,37 @@
+const getCatogriedProducts = (categories, products) => {
+
+    const categoriedProductReducer = (acc, currItem) => {
+        const productsArray = products.filter((item) => item.categoryName === currItem);
+        acc = acc.concat(productsArray);
+        return acc;
+    }
+
+    if(categories.length === 0) {
+        return products
+    }
+    else{
+        return categories.reduce(categoriedProductReducer, [])
+    }
```

Comment:
Suggestion: 
no need of else. You can just return **categories.reduce(categoriedProductReducer, [])**.
For checking the length, you can use not (!) operator.
Just a suggestion. Nothing wrong with the existing code. Please ignore it if not needed.

Reply:
Actually I didnt know I can do that, thanks !!